src/test/java/spoon/test/refactoring/ChangeVariableNameTest.java

```diff
@@ -0,0 +1,244 @@
+		}
+		if(renameShouldPass) {
+			rollback(targetVariable, originName);
+		}
+		assertEquals(originName, targetVariable.getSimpleName());
+	}
+	
+	private void rollback(CtLocalVariable<?> targetVariable, String originName) {
+		String newName = targetVariable.getSimpleName();
+		ChangeLocalVariableName refactor = new ChangeLocalVariableName();
+		refactor.setTarget(targetVariable);
+		//rollback changes
+		refactor.setNewName(originName);
+		try {
+			refactor.refactor();
+		} catch(SpoonException e) {
+			throw new AssertionError(getParentMethodName(targetVariable)+" Rename of \""+originName+"\" to \""+newName+"\" passed, but rename back to \""+originName+"\" failed", e);
+		}
+	}
+
+	private void printModelAndTestConsistency(String refactoringDescription) {
+//		 1) print modified model,
+		try {
+			launcher.getModelBuilder().generateProcessedSourceFiles(OutputType.CLASSES);
+		} catch (Throwable e) {
+			new AssertionError("The printing of java sources failed after: "+refactoringDescription, e);
+		}
+		
+//		 2) build it
+		try {
+//			launcher.getModelBuilder().compile(SpoonModelBuilder.InputType.FILES);
+			launcher.getModelBuilder().compile(SpoonModelBuilder.InputType.CTTYPES);
+		} catch (Throwable e) {
+			new AssertionError("The compilation of java sources in "+launcher.getEnvironment().getBinaryOutputDirectory()+" failed after: "+refactoringDescription, e);
+		}
+//		 3) create instance using that new model and test consistency
+		try {
+//			varRenameClass.newInstance();
+			TestClassloader classLoader = new TestClassloader();
+			Class testModelClass = classLoader.loadClass(VariableRename.class.getName());
+			testModelClass.newInstance();
+		} catch (Throwable e) {
+			throw new AssertionError("The model validation of code in "+launcher.getEnvironment().getBinaryOutputDirectory()+" failed after: "+refactoringDescription, e);
+		}
+	}
+	
+	private class TestClassloader extends URLClassLoader {
+		TestClassloader() throws MalformedURLException {
+			super(new URL[] { new File(launcher.getEnvironment().getBinaryOutputDirectory()).toURL()}, ChangeVariableNameTest.class.getClassLoader());
+		}
+
+		@Override
+		public Class<?> loadClass(String s) throws ClassNotFoundException {
+			try {
+				return findClass(s);
+			} catch (Exception e) {
+				return super.loadClass(s);
+			}
+		}
+	}
+	
+	
+	private String getParentMethodName(CtElement ele) {
+		CtMethod parentMethod = ele.getParent(CtMethod.class);
+		CtMethod m;
+		while(parentMethod!=null && (m=parentMethod.getParent(CtMethod.class))!=null) {
+			parentMethod = m;
+		}
+		if(parentMethod!=null) {
+			return parentMethod.getParent(CtType.class).getSimpleName()+"#"+parentMethod.getSimpleName();
+		} else {
+			return ele.getParent(CtType.class).getSimpleName()+"#annonymous block";
+		}
+	}
+	
+
+	@Test
+	public void testRefactorWithoutTarget() throws Exception {
+		

```

Comment:
contract: a name refactoring where the given variable name does not exist results in a SpoonException